src/test/java/seedu/recipe/testutil/TypicalRecipes.java

```diff
@@ -232,6 +199,39 @@ public class TypicalRecipes {
                             + "if you want heat."),
                     new Step("Assemble together and enjoy!"))).build();
 
+    public static final String CACIO_STRING = String.format(
+            "%s;\nPortion: %s;\nDuration: %s;\nTags: %s;\n"
+                    + "Ingredients: %s;\nSteps: %s",
+            CACIO_NAME, CACIO_PORTION, CACIO_DURATION, (
+            (Supplier<String>) () -> {
+                StringBuilder out = new StringBuilder();
+                ArrayList<Tag> tags = new ArrayList<>(CACIO_TAGS);
+                tags.sort(Comparator.comparing(tag -> tag.tagName));
+                for (Tag tag : tags) {
+                    out.append(tag.toString());
+                }
+                return out.toString();
+            }).get(), (
+            (Supplier<String>) () -> {
+                StringBuilder out = new StringBuilder();
+                for (IngredientBuilder i : CACIO_INGREDIENTS) {
+                    out.append(i.toString())
+                            .append(",\n");
+                }
+                return out.toString();
+            }).get(), (
+            (Supplier<String>) () -> {
+                StringBuilder out = new StringBuilder();
+                for (int i = 0; i < CACIO_STEPS.size(); i++) {
+                    out.append(i + 1)
+                            .append(". ")
+                            .append(CACIO_STEPS.get(i).toString())
+                            .append(",\n");
+                }
+                return out.toString();
```

Comment:
This chunk can be divided into 3 separate helper functions instead of using anonymous supplier instances, with the main code just calling the 3 helper functions. This is preference-based and just makes it a bit more readable, so its up to you.

Reply:
fixed